### ISSUE TITLE
Skip build docker image prompt in `prefect deploy` when `build` is explicitly set to `null` for that deployment

### ISSUE THREAD

**Issue by taylor-curran** [PrefectHQ/prefect#10274]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

This was happening with schedules too #9823. I'd say its lower priority since `prefect --no-prompt deploy --name a-local-docker` makes this "bug" irrelevant. 

### Reproduction

```python3
My Prefect YAML


# Welcome to your prefect.yaml file! You can you this file for storing and managing
# configuration for deploying your flows. We recommend committing this file to source
# control along with your flow code.

# Generic metadata about this project
name: prefect-patterns
prefect-version: 2.10.17

# build section allows you to manage and build docker images
build:
- prefect.deployments.steps.run_shell_script:
    id: get-commit-hash
    script: git rev-parse --short HEAD
    stream_output: false
- prefect_docker.deployments.steps.build_docker_image:
    requires: prefect-docker>=0.3.0
    image_name: taycurran/test-projects-june11
    tag: '{{ get-commit-hash.stdout }}'
    dockerfile: auto
    push: true

# push section allows you to manage if and how this project is uploaded to remote locations
push:

# pull section allows you to provide instructions for cloning this project in remote locations
pull:
- prefect.deployments.steps.git_clone:
    repository: https://github.com/taylor-curran/prefect-patterns.git
    branch: main

definitions:
  tags:
  - subflow-patterns
  work_pool: &local_docker
    name: local-docker
    job_variables:
      image: '{{ image }}'

# the deployments section allows you to provide configuration for deploying flows
deployments:
- name: base-deployment
  version:
  tags:
  - '{{ get-commit-hash.stdout }}'
  description:
  entrypoint: 
    flows/composite_flow_patterns/tasks_subflows_models/child_flows_sync.py:child_flow_a
  parameters: {}
  work_pool: *local_docker
  schedule:
  build:
  - prefect_docker.deployments.steps.build_docker_image:
    requires: prefect-docker>=0.3.0
    image_name: taycurran/test-projects-june11
    tag: '{{ get-commit-hash.stdout }}'
    dockerfile: auto
    push: true 
- name: a-local-docker
  version:
  tags:
  - docker
  - child
  - '{{ get-commit-hash.stdout }}'
  description:
  entrypoint: 
    flows/composite_flow_patterns/tasks_subflows_models/child_flows_sync.py:child_flow_a
  parameters: {}
  work_pool: *local_docker
  schedule:
  build: null


prefect deploy --name a-local-docker
```

### Error

```python3
(prfp) ➜  prefect-patterns git:(main) ✗ prefect deploy --name a-local-docker 
? Would you like to build a custom Docker image for this deployment? [y/n] (n): n
```
```


### Versions

```Text
Version:             2.11.0
API version:         0.8.4
Python version:      3.9.16
Git commit:          eeb9e219
Built:               Thu, Jul 20, 2023 4:34 PM
OS/Arch:             darwin/arm64
Profile:             prefect-patterns
Server type:         cloud
```
